src/components/Comment.jsx

```diff
@@ -1,24 +1,81 @@
-import { useContext } from "react";
+import { useContext, useState, useEffect } from "react";
 import { UserContext } from "../contexts/UserContext";
+import { deleteComment } from "../utils/apiRequests";
 
-const Comment = ({ comment }) => {
+const Comment = ({ comment, setComments }) => {
+  const [isUser, setIsUser] = useState(true);
   const userValue = useContext(UserContext);
   const posted = comment.created_at
     ? comment.created_at.split(/[-T:.]/)
     : ["just now."];
+
+  useEffect(() => {
+    if (comment.author === userValue.loggedInUser.username) {
+      setIsUser(true);
+    } else {
+      setIsUser(false);
+    }
+  }, [userValue.loggedInUser.username, comment.author]);
+
+  const deleteHandler = () => {
+    setComments((currComments) => {
+      const copy = [...currComments];
+      const index = copy.findIndex((element) => {
+        return element.comment_id === comment.comment_id;
+      });
+      copy.splice(index, 1);
+      return copy;
+    });
+    deleteComment(comment.comment_id).catch(() => {
+      alert("your comment did not delete. try again");
```

Comment:
They won't be able to try again as you've already removed it from the page. You'll need to put it back if you want them to be able to do that

Reply:
I've had a go at fixing this by adding a state with the deleted comment in and splicing it back into the array in the catch block. The state doesn't seem to want to update though, can you see why?

Reply:
So you probably don't need the extra state here as you have it in closure. (currComments[index]).

Depending on how you test this you might run into async issues with the state not having updated yet. e.g. no network will fail immediately and React might batch the setStates causing it to bug out.

Use the values that you put into state directly and it should behave better.

Reply:
I can't work out how to get the deleted comment out of the setComments block on line 13. Everything I've tried causes an error. Could I use `const deletedComment = []` outside of the block and push it?

Reply:
I think this works now.